### PR TITLE
Bugfix/1952 gethostbyname

### PR DESCRIPTION
**Pull request by omusil24**
Rewritten all `q_gethostbyname` functions to use `QoreAddrInfo` internally and
rewritten `q_gethostbyaddr` and `q_gethostbyaddr_string` using `getnameinfo(3)`.
`q_gethostbyaddr_to_hash` was left as it is as it's not possible to easily rewrite it using `getnameinfo` - this is because `q_gethostbyaddr_to_hash` is supposed to return the hostname + all aliases but `getnameinfo` returns only the hostname.

@pvanek @davidnich is this okay?

### CODE REVIEW

File: doxygen/lang/900_release_notes.dox.tmpl [hunk 1]
```diff
@@ -253,6 +253,7 @@
     - fixed a bug in overloaded call variant matching where missing arguments were counted towards the match (<a href="https://github.com/qorelanguage/qore/issues/1897">issue 1897</a>)
     - fixed many bugs where parse-time errors could be reported at an incorrect source location; parse-time error location reporting has been completely overhauled and reimplemented for correctness (<a href="https://github.com/qorelanguage/qore/issues/1930">issue 1930</a>)
     - fixed a bug where code signatures would accept parameter variables without \c "$" signs even when @ref allow-bare-refs "%allow-bare-refs" was not in effect (<a href="https://github.com/qorelanguage/qore/issues/1941">issue 1941</a>)
+    - rewritten the Qore functions \ref gethostbyname(), \ref gethostbyname_long() and \ref gethostbyaddr() to use standard C functions \c getaddrinfo(3) and \c getnameinfo(3) internally instead of the deprecated \c gethostbyname(3) and \c gethostbyaddr(3) (<a href="https://github.com/qorelanguage/qore/issues/1952">issue 1952</a>)
```
Comment (davidnich) [inline doxygen/lang/900_release_notes.dox.tmpl]: please change to `rewrote %Qore functions`  ... also why use `\ref` when all the documentation everywhere is using `@ref`?

**Review by davidnich — CHANGES_REQUESTED** [review]
@omusil24 I only have one minor cosmetic request in the release note - looks good

File: doxygen/lang/900_release_notes.dox.tmpl (hunk 1 above)
Comment (omusil24) [reply doxygen/lang/900_release_notes.dox.tmpl]: Ok... I didn't notice that at the time... I'll rewrite it.

**Review by davidnich — APPROVED** [review]

**Review by pvanek — APPROVED** [review]
ciao. It looks good. I have only one comment (not an issue). The whitespace changes are quite aggressive. See eg. f9ff8e93af21050d6bdb653464bef6d998c29622 there is almost impossible to find if there was a real change or just a "* moving by one space" everywhere. 

It brings back the problem of code formatting. What about to make an one global "astyle" reformat of all source codes in one commit. Then we can have just real changes in reviews...

**Comment by davidnich** [comment]
@pvanek ondra gave the solution to the whitespace issue - just append ?w=1 to the URI path and you will get diffs without whitespace changes

note that if we make whitespace fixes, we should also do the 3 spaces -> 4 spaces change as well - I have no problem with it if we kill both birds with the same stone

**Comment by pvanek** [comment]
w=1 is not a solution, just a workaround. These formatting changes should not be part of real changes. That's all ;)

**Comment by omusil24** [comment]
That formatting change would be good to do finally... We just have to get rid of all open branches other than 0.8.11.1, 0.8.12, and develop...

**Comment by davidnich** [comment]
@pvanek your comment that whitespace changes should not be a part of real changes is not our current policy as formulated and agreed back in the Ondra days.

The rationale behind allowing whitespace fixes in any pull request is that it allows improvements to be made without making it harder to do the review.  This is true because we have the workaround to filter out whitespace changes (i.e. `?w=1`), and this is the reason that we explicitly agreed to allow whitespace changes in pull requests quite some time ago.

By forbidding whitespace changes in pull requests, we would make it harder to make whitespace changes by forcing them to be in a separate pull request and therefore increase the effort needed to make such improvements which is counter-productive when we have such resources to move things forward.

If you want to solve it by making a pull request to fix all the whitespace changes in one go, no problem, please do it.

**Pull request merged**